Classes/Backend/DynamicFlexForm.php

```diff
@@ -162,6 +162,10 @@ public function parseDataStructureByIdentifierPreProcess(array $identifier)
         } else {
             $dataSourceIdentity = $identifier['tableName'] . '.' . $identifier['fieldName'] . ':' . $identifier['uid'];
         }
+        if(!isset(static::$generatedDataSources[$dataSourceIdentity])) {
```

Comment:
This should already be filled by the `getDataStructureIdentifierPreProcess` which gets called by FormEngine. Clearly the identifier was generated by the pre process method, because it contains the record UID. This means that the DS should already have been generated and stored in the internal cache.

Calling the pre-processing method from here is definitely not the way we want to go - the problem should be solved in the pre process method, if this method for some reason fails to generate a proper DS when the UID is provided. Such behaviour is definitely not expected, but I don't currently know what the cause might be. The pre-processing method should be called consistently so it is probably some flaw in the method's logic as implemented in this class...